src/amp/components/Experiment.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+
+export interface ExperimentModel {
+    [key: string]: {
+        sticky: boolean;
+        consentNotificationId?: string;
+        variants: {
+            [key: string]: number; // We do this for compatibility with ExperimentModel via ExperimentConfig
+        };
+    };
+}
+
+interface StyledVariants { [key: string]: { proportion: number; style: string;}; }
+
+export interface ExperimentFullConfig {
+    [key: string]: {
+        sticky: boolean;
+        consentNotificationId?: string;
+        variants: StyledVariants;
+    };
+}
+
+
+export interface ExperimentStyle {
+    [key: string]: {[key: string]: string};
+}
+
+const extractStyleFromVariants = (variants: StyledVariants): [{[key: string]: number}, {[key: string]: string}] =>
+    Object.entries(variants).reduce(
+    (acc, [variant, config])=>{
+        const {proportion, style} = config;
+        acc = [{...acc[0], ...{[variant]: proportion}}, {...acc[1], ...{[variant]: style}}];
+        return acc
+    },
+    [{}, {}]
+);
+export const extractModelAndStyle = (fullConfig: ExperimentFullConfig): [ExperimentModel, ExperimentStyle] => {
```

Comment:
Can you elaborate on/document this function?

Reply:
Where should this be done? In the module or in external documentation?

Reply:
Refactored this function into two functions which should hopefully be more understandable.